File: edupilot_api/EduPilot.Api/DTOs/StudentRegisterDTO.cs
using Newtonsoft.Json.Serialization;
using System.ComponentModel.DataAnnotations;

namespace EduPilot.Api.Models
{
    public class StudentRegisterDTO
    {
        [Required]
        public string FirstName { get; set; }
        public string? MiddleName { get; set; }
        [Required]
        public string LastName { get; set; }
        [Required]
        [Range(1, 12, ErrorMessage = "Sınıf 1 ile 12 arasında olmalıdır.")]
        public int Grade { get; set; }
        [Required]
        [EmailAddress]
        public string Email { get; set; }
        [Required]
        [DataType(DataType.Password)]
        public string Password { get; set; }
        [Required]
        [DataType(DataType.PhoneNumber)]
        public string PhoneNumber { get; set; }
        [Required]
        public string Avatar { get; set; }
        [Required]
        public List<Guid> FavoriteLessons { get; set; }
        public int? SupervisorUniqueCode { get; set; }
        public string? SupervisorName { get; set; }
    }
}

File: edupilot/lib/models/dtos/student_register_dto.dart
class StudentRegisterDTO {
  final String firstName;
  final String? middleName;
  final String lastName;
  final int grade;
  final String email;
  final String password;
  final String phoneNumber;
  final String avatar;
  final List<String> favoriteLessons;
  final String? supervisorName;
  final int? supervisorUniqueCode;

  const StudentRegisterDTO({
    required this.firstName,
    this.middleName,
    required this.lastName,
    required this.grade,
    required this.email,
    required this.password,
    required this.phoneNumber,
    required this.avatar,
    required this.favoriteLessons,
    this.supervisorName,
    this.supervisorUniqueCode,
  });

  factory StudentRegisterDTO.fromJson(Map<String, dynamic> json) =>
      StudentRegisterDTO(
        firstName: json['firstName'] as String,
        middleName: json['middleName'] as String?,
        lastName: json['lastName'] as String,
        grade: json['grade'] as int,
        email: json['email'] as String,
        password: json['password'] as String,
        phoneNumber: json['phoneNumber'] as String,
        avatar: json['avatar'] as String,
        favoriteLessons:
            (json['favoriteLessons'] as List<dynamic>).map((e) => e as String).toList(),
        supervisorName: json['supervisorName'] as String?,
        supervisorUniqueCode: json['supervisorUniqueCode'] as int?,
      );

  Map<String, dynamic> toJson() => {
        'name': firstName,
        'middleName': middleName,
        'lastName': lastName,
        'grade': grade,
        'email': email,
        'password': password,
        'phoneNumber': phoneNumber,
        'avatar': avatar,
        'favoriteLessons': favoriteLessons,
        'supervisorName': supervisorName,
        'supervisorUniqueCode': supervisorUniqueCode,
      };
}